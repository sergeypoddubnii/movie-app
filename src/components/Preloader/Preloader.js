import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import Box from '@material-ui/core/Box';
import PropTypes from 'prop-types';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    '& > * + *': {
      marginLeft: theme.spacing(2),
    },
    transform: 'translate(-50%, -50%)',
    top: '50%',
    left: '50%',
    position: 'absolute',
  },
}));

const Preloader = ({ isLoading }) => {
  const classes = useStyles();
  return isLoading ? (
    <Box
      width="100%"
      height="100%"
      position="absolute"
      top="0"
      left="0"
      bgcolor="rgba(0,0,0,0.2)"
      zIndex="100"
    >
      <div className={classes.root}>
        <CircularProgress />
      </div>
    </Box>
  ) : null;
};

Preloader.propTypes = {
  isLoading: PropTypes.bool,
};

export default Preloader;
