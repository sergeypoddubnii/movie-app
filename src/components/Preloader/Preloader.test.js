import { shallow } from 'enzyme';
import React from 'react';
import Preloader from './Preloader';

describe('Preloader Test', () => {
  it('should render with props', () => {
    const component = shallow(<Preloader isLoading={true} />);
    expect(component).toMatchSnapshot();
  });

  it('should render without props', () => {
    const component = shallow(<Preloader />);
    expect(component).toMatchSnapshot();
  });
});
