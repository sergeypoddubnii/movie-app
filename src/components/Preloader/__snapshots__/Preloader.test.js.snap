// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader Test should render with props 1`] = `
<Styled(MuiBox)
  bgcolor="rgba(0,0,0,0.2)"
  height="100%"
  left="0"
  position="absolute"
  top="0"
  width="100%"
  zIndex="100"
>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(CircularProgress)) />
  </div>
</Styled(MuiBox)>
`;

exports[`Preloader Test should render without props 1`] = `""`;
